contracts/AelinUpFrontDeal.sol

```diff
@@ -0,0 +1,299 @@
+
+        if (nftCollectionRules.length > 0) {
+            // if the first address supports punks or 721, the entire pool only supports 721 or punks
+            if (
+                nftCollectionRules[0].collectionAddress == CRYPTO_PUNKS ||
+                NftCheck.supports721(nftCollectionRules[0].collectionAddress)
+            ) {
+                for (uint256 i = 0; i < nftCollectionRules.length; i++) {
+                    require(
+                        nftCollectionRules[i].collectionAddress == CRYPTO_PUNKS ||
+                            NftCheck.supports721(nftCollectionRules[i].collectionAddress),
+                        "can only contain 721"
+                    );
+                    nftCollectionDetails[nftCollectionRules[i].collectionAddress] = nftCollectionRules[i];
+                    emit PoolWith721(
+                        nftCollectionRules[i].collectionAddress,
+                        nftCollectionRules[i].purchaseAmount,
+                        nftCollectionRules[i].purchaseAmountPerToken
+                    );
+                }
+                hasNftList = true;
+            }
+            // if the first address supports 1155, the entire pool only supports 1155
+            else if (NftCheck.supports1155(nftCollectionRules[0].collectionAddress)) {
+                for (uint256 i = 0; i < nftCollectionRules.length; i++) {
+                    require(NftCheck.supports1155(nftCollectionRules[i].collectionAddress), "can only contain 1155");
+                    nftCollectionDetails[nftCollectionRules[i].collectionAddress] = nftCollectionRules[i];
+
+                    for (uint256 j = 0; j < nftCollectionRules[i].tokenIds.length; j++) {
+                        nftId[nftCollectionRules[i].collectionAddress][nftCollectionRules[i].tokenIds[j]] = true;
+                    }
+                    emit PoolWith1155(
+                        nftCollectionRules[i].collectionAddress,
+                        nftCollectionRules[i].purchaseAmount,
+                        nftCollectionRules[i].purchaseAmountPerToken,
+                        nftCollectionRules[i].tokenIds,
+                        nftCollectionRules[i].minTokensEligible
+                    );
+                }
+                hasNftList = true;
+            } else {
+                revert("collection is not compatible");
+            }
+        }
+
+        // deposit underlying token logic
+        // check if the underlyingDealAmount is more than 0,
+        // if yes, transfer it to this contract and store it in `currentDealTokenTotal` and add it
+        if (_underlyingDealTokenAmount > 0) {
+            currentDealTokenTotal += _underlyingDealTokenAmount;
+            IERC20(_dealData.underlyingDealToken).transferFrom(_sponsor, address(this), _underlyingDealTokenAmount);
+            if (_underlyingDealTokenAmount >= _dealData.underlyingDealTokenTotal) {
```

Comment:
we dont need both of these _underlyingDealTokenAmount >= _dealData.underlyingDealTokenTotal
in this case we only need one value for the deal total. we need a few workflows around depositing the funds and having holder and sponsor clarified. I will post a comment about this later.